### PR TITLE
EOS-23169: Factory Reset - Call Hare init stage in Hare 'reset' stage

### PR DESCRIPTION
**Pull request by supriyachavan4398**
For motr reset, to restart the cluster, hare init needs to be called.
So Hare reset needs to call the 'Init' stage.

Solution:
Updated hare.reset.conf.tmpl.1-node and hare.reset.conf.tmpl.3-node.
Call 'Init' stage into Hare reset stage.
Addressed review comments on PR #1727 and removed /var/motr/hax , /etc/sysconfig/m0d-*
and /etc/sysconfig/s3server-* files in cleanup stage with pre-factory arguement.

Signed-off-by: Supriya Yadav <supriya.s.chavan@seagate.com>

### CODE REVIEW

**Comment by cortx-admin** [comment]
Can one of the admins verify this patch?

File: provisioning/miniprov/hare_mp/main.py [hunk 1]
```diff
@@ -282,7 +282,7 @@ def test_IVT(args):
 def reset(args):
     try:
         rc = 0
-
+        init(args)
```
Comment (knekrasov) [inline provisioning/miniprov/hare_mp/main.py:285]: 1. `init` may exit. Please refactor `init()` function so it will not stop the whole application unexpectedly. Without it it is dangerous to compose functions. Putting `exit()`s within the functions was initially not the best idea.
2. Please add a comment explaining why this init is required.

Comment (supriyachavan4398) [reply provisioning/miniprov/hare_mp/main.py:285]: This is regarding the RESET phase of the mini provisioner.
In motr reset, motr clean up the motr and IO tests generated data. But to restart the cluster, hare init needed to be called.
As a part of motr reset, we need to mkfs and start m0d services, motr wants hare to start it through hare init.
So its not actually cluster start but it is services start.

Comment (supriyachavan4398) [reply provisioning/miniprov/hare_mp/main.py:288]: Addressed.

**Comment by knekrasov** [comment]
Please address the codacy issues. It seems like part of them are related to formatting. Please consider using automated tool to format the code - [`yapf`](https://github.com/google/yapf)

**Comment by supriyachavan4398** [comment]
> Please address the codacy issues. It seems like part of them are related to formatting. Please consider using an automated tool to format the code - [`yapf`](https://github.com/google/yapf)

Addressed and refactor all the codacy issues.

**Comment by supriyachavan4398** [comment]
VM Deployment job : http://eos-jenkins.colo.seagate.com/job/Cortx-Deployment/job/VM-Deployment-R2-1Node/1884/
@vaibhavparatwar  @mssawant

File: provisioning/miniprov/hare_mp/main.py [hunk 2]
```diff
@@ -657,11 +674,11 @@ def main():
                                nargs='?',
                                help='Target directory; defaults to /tmp/hare.')
 
-    add_subcommand(subparser,
-                   'reset',
-                   help_str='Resets temporary Hare data and configuration',
-                   handler_fn=reset,
-                   config_required=False)
+    add_file_argument(
```
Comment (SwapnilGaonkar7) [inline provisioning/miniprov/hare_mp/main.py:677]: Why this change is needed?

Comment (supriyachavan4398) [reply provisioning/miniprov/hare_mp/main.py:677]: In the reset stage, we are calling the `init` stage which requires the `--file` argument. For this, we have added these changes here.

Comment (supriyachavan4398) [reply provisioning/miniprov/hare_mp/main.py:677]: Addressed.

**Review by knekrasov — APPROVED** [review]
I took a liberty to fix test_templates within your PR to verify that your patch doesn't break mini provisioner.

**Review by SwapnilGaonkar7 — APPROVED** [review]

**Comment by SwapnilGaonkar7** [comment]
@supriyachavan4398 Make sure deployment is passing

**Review by mssawant — APPROVED** [review]
